### DISCUSSION TITLE
Support of Angular

### DISCUSSION THREAD

**Discussion by dirk-ecker** [vitest-dev/vitest#3024] [Ideas]
Does Vitest support angular component testing?
Would be really nice.
How is it done?

**Comment by kytta** [comment]
Jasmine, Jest, Vitest — all these libraries are mostly test _runners_, and in case of Angular, neither of those is used to actually bootstrap the app. That it, you use the same approach as you do with Jasmine (`TestBed`, `ComponentFixture`, etc.), but replace its specific `expect`, `it`, `spyOn`, etc. with the ones of Vitest.

I myself just started looking into the possibility of this, so I can't provide examples. Two links I've found that might be helpful:

- https://github.com/thymikee/jest-preset-angular — here's how you would test your components with Jest. Do this, then follow Vitest's migration guide.
- https://analogjs.org/docs/features/testing/vitest — Analog is a meta-framework for Angular, but this setup could be used for Vanilla Angular too, I suppose

**Reply by brandonroberts** [reply]
That's correct. You can use the guide to set up Vitest with a vanilla Angular application also
